ci-operator/step-registry/baremetalds/assisted/setup/baremetalds-assisted-setup-ref.yaml

```diff
@@ -19,6 +19,8 @@ ref:
     env: ASSISTED_AGENT_IMAGE
   - name: "assisted-installer-controller"
     env: ASSISTED_CONTROLLER_IMAGE
+  - name: "assisted-installer-controller"
+    env: PROVIDER_IMAGE
```

Comment:
It seems reasonable as long as we:
* start now adding it anywhere needed. No reason we wouldn't add the dependency to all our owned files. release-controllers and external repos can be handled separately 
* add a comment here explaining that this is deliberate and not a mistake